folly/memory/UninitializedMemoryHacks.h

```diff
@@ -110,6 +130,11 @@ inline void resizeWithoutInitialization(
       s.reserve(n);
     }
     detail::unsafeStringSetLargerSize(s, n);
+#if defined(_MSC_VER) && _MSC_VER >= 1938 && defined(_INSERT_STRING_ANNOTATION)
+    if (_Asan_string_should_annotate && s.capacity() > std::string().capacity())
+      __sanitizer_annotate_contiguous_container(&s[0], &s[0] + s.capacity() + 1,
```

Comment:
We already wrap some of the sanitizer interface. We exclusively use the wrappers.

Reply:
Expect in specific cases. For example, the existing calls to `__sanitizer_annotate_contiguous_container` below are exactly as libc++ does, located in the per-library implementation for libc++.